multithreading/protocols/smbus_stream_reader.py

```diff
@@ -13,5 +20,38 @@ def __init__(self, lock, data_queue, log_queue, config):
             config (dict): The configuration for the sensor (I2C device id).
         """
         super().__init__(lock, data_queue, log_queue, config)
-        revision = ([l[12:-1] for l in open('/proc/cpuinfo','r').readlines() if l[:8]=="Revision"]+['0000'])[0]
-        self.bus = smbus.SMBus(config["new_rev_i2c_id"] if int(revision, 16) >= 4 else config["old_rev_i2c_id"])
\ No newline at end of file
+
+        if SMBusStreamReader._bus is None:
+            revision = ([l[12:-1] for l in open('/proc/cpuinfo','r').readlines() if l[:8]=="Revision"]+['0000'])[0]
+            SMBusStreamReader._bus = smbus2.SMBus(SMBUS_NEW_ID if int(revision, 16) >= 4 else SMBUS_OLD_ID)
```

Comment:
magic numbers everywhere, 12, -1, 8, 16, 4

Reply:
I know this is lazy, but since we're only using one Raspberry Pi 4, I'll remove support of OLD/NEW SMBus ids. I'm not sure how it worked anyway